EjercicioClases/Controller/MainController.pm

```diff
@@ -49,99 +43,40 @@ sub startApp{
 }
 
 sub createSeller{
-    my @sellerData = View->inputForm();
-    if(isDuplicatedId($sellerData[0]) eq 'no' && isValidEmail($sellerData[2]) eq 'yes'){
-        my $seller = Model::User::Seller->new($sellerData[0], $sellerData[1], $sellerData[2], $sellerData[3], $sellerData[4]);
-        Controller::DatabaseController->saveSeller($seller);
-        #saveSeller($seller);
+    my @sellerData = View::View->inputForm();
+    my $seller = Model::User::Seller->new($sellerData[0], $sellerData[1], $sellerData[2], $sellerData[3], $sellerData[4]);
+    if($seller->isDuplicatedId($seller->getId) eq 'no' && $seller->isValidEmail($seller->getEmail)){
+        $seller->save($seller);
     }
 }
 
 sub createBuyer{
-    my @buyerData = View->inputForm();
-    if(idDuplicatedId($buyerData[0]) eq 'no' && isValidEmail($buyerData[2]) eq 'yes'){
-        my $buyer = Model::User::Buyer->new($buyerData[0], $buyerData[1], $buyerData[2]);
-        Controller::DatabaseController->saveBuyer($buyer); #hash with key table name
+    my @buyerData = View::View->inputForm();
+    my $buyer = Model::User::Buyer->new($buyerData[0], $buyerData[1], $buyerData[2]);
+    if($buyer->isDuplicatedId($buyer->getId) eq 'no' && $buyer->isValidEmail($buyer->getEmail)){
+        $buyer->save($buyer);
     }
 }
 
 sub createProduct{
-    my @productData = View->inputForm();
-    my $product = Model::Product->new($productData[0], $productData[1], $productData[2], 'yes');
-    Controller::DatabaseController->saveProduct($product);
+    my @productData = VView::iew->inputForm();
+    Model::Product->save(Model::Product->new($productData[0], $productData[1], $productData[2], 'yes'));
 }
 
 sub createTransaction{
-    my @transactionData = View->inputForm();
-    my $productToSell = Controller::DatabaseController->retrieveProductBySku($transactionData[1]);
+    my @transactionData = View::View->inputForm();
+    my $productToSell = Model::Product->getProductBySku($transactionData[1]);
     if($productToSell->getStatus eq 'yes'){
```

Comment:
P=lease change the 'getStatus' function name to something more related to the actual attribute. For example 'isAvailable', 'hasStock' etc.